pytorch_lightning/trainer/supporters.py

```diff
@@ -263,16 +308,21 @@ def __init__(self, loaders: Any, mode='min_size'):
         if self.mode == 'max_size_cycle':
             self._wrap_loaders_max_size_cycle()
 
+    @staticmethod
+    def extract_dataset(loader):
```

Comment:
this does not work with all loaders, since we strictly don't restrict people to use pytorch data loaders but any iterators (and they don't always have a dataset attribute. Also one may want to use different batch sizes here, so it would be better to actually wrap the iterators only and don't rely on dataset wrapping.

Reply:
The reason why I add `dataset` here is in the second reply.